crates/storage/src/merkle/column.rs

```diff
@@ -42,21 +51,30 @@ where
     TC: strum::EnumCount + AsU32,
 {
     /// The total count of variants in the enum.
-    /// Since we have two columns for each table column and one for the merkle data,
-    /// we have to multiply the count of the table columns by 2 and add one for the merkle metadata.
-    pub const COUNT: usize = TC::COUNT * 2 + 1;
+    pub const COUNT: usize = <Self as strum::EnumCount>::COUNT;
 
     /// The start of the merkle data columns.
     pub const MERKLE_DATA_COLUMNS_START: u32 = u16::MAX as u32;
 
+    /// The merkle metadata column
+    pub const MERKLE_METADATA_COLUMN: u32 = {
+        assert!(Self::COUNT <= u32::MAX as usize);
+        // this is fine, because we already performed an assertion above
+        // see https://github.com/rust-lang/rust-clippy/issues/9613
+        #[allow(clippy::cast_possible_truncation)]
+        let column_index = (Self::COUNT as u32).saturating_sub(1);
+        assert!(column_index > 0);
+        column_index
+    };
```

Comment:
If you do it like this, it means that column id depends on the `Self::COUNT`. So if you add a new column, it will override the metadata column. Metadata column always should be stable regardless of the number of columns in the type.

Reply:
addressed in aa438a2295